madrid/bicimad.py

```diff
@@ -0,0 +1,144 @@
+import requests
+import pandas as pd
+
+from madrid.auth_credentials import auth_bicimad
+
+
+def shout(func):
+    def wrapper(*args, **kwargs):
+        print(f'Executing {func.__name__}')
+        return func(*args, **kwargs)
+
+    return wrapper
+
+
+class BiciMad:
+    """
+    Class in charge of connecting to BiciMad API and querying information
+    Documentation here: https://apidocs.emtmadrid.es/
+    """
+    url_login = 'https://openapi.emtmadrid.es/v1/mobilitylabs/user/login/'
+    url_logout = 'https://openapi.emtmadrid.es/v1/mobilitylabs/user/logout/'
+    url_who_am_i = 'https://openapi.emtmadrid.es/v1/mobilitylabs/user/whoami/'
+    url_get_all_stations_info = 'https://openapi.emtmadrid.es/v1/transport/bicimad/stations/'
+
+    stations_keys = ['id', 'number', 'activate', 'total_bases', 'dock_bikes', 'free_bases', 'reservations_count']
```

Comment:
Are these all the available columns or are there more? For example, for bcn there is a column which shows the five closest stations to each particular station. Just wondering whether you selected only these or whether there are more.

Reply:
There are some more, which are in principle static (not changing with time: lat, lon), but as you say, why not save them in ddbb and just discard them in the future if not needed.